### PR TITLE
Add gui combobox missing functions

### PR DESCRIPTION
**Pull request by FileEX**
**Syntax**
```lua
int guiComboBoxGetItemCount(element theElement)
bool guiComboBoxSetOpen(element theElement, bool state)
bool guiComboBoxIsOpen(element theElement)
```

guiComboBoxIsOpen return boolean, nil if bad argument
guiComboBoxSetOpen return boolean

### CODE REVIEW

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 1]
```diff
@@ -5189,6 +5189,80 @@ bool CStaticFunctionDefinitions::GUIComboBoxSetItemText(CClientEntity& Entity, i
     return false;
 }
 
+int CStaticFunctionDefinitions::GUIComboBoxGetItemCount(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxGetItemCount(**iter))
+
+    // Are we a CGUI Element?
+        if (IS_GUI(&Entity))
```
Comment (jushar) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Please fix the indentation here and below.

File: Client/mods/deathmatch/logic/luadefs/CLuaGUIDefs.cpp [hunk 2]
```diff
@@ -3887,6 +3892,71 @@ int CLuaGUIDefs::GUIComboBoxSetItemText(lua_State* luaVM)
...
+}
+
+int CLuaGUIDefs::GUIComboBoxSetState(lua_State* luaVM)
+{
+    // int guiComboBoxSetState( element comboBox, int state)
+    CClientGUIElement* comboBox;
+    int state;
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData<CGUIComboBox>(comboBox);
+    argStream.ReadNumber(state);
```
Comment (jushar) [inline Client/mods/deathmatch/logic/luadefs/CLuaGUIDefs.cpp]: I'd personally prefer an enum string here (and ofc return a string in `guiComboxGetState`). I'd like to hear other opinions though.

**Or, an even better solution - as you only distinguish two states - is to rename the function to `guiComboxIsOpen` and make it return a _boolean_.**

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 3]
```diff
@@ -5189,6 +5189,80 @@ bool CStaticFunctionDefinitions::GUIComboBoxSetItemText(CClientEntity& Entity, i
...
+
+int CStaticFunctionDefinitions::GUIComboBoxGetState(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxGetState(**iter))
+
+        // Are we a CGUI Element?
+        if (IS_GUI(&Entity))
+        {
+            CClientGUIElement& GUIElement = static_cast<CClientGUIElement&>(Entity);
+            int comboState;
```
Comment (jushar) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Tip: Try reducing the scope of variables as much as possible, i.e. move `comboState` into the if block or avoid creating it entirely by directly returning from the `vs` / `!vs` - if blocks.

**Review by jushar — CHANGES_REQUESTED** [review]

**Comment by FileEX** [comment]
Everything fixed

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 4]
```diff
@@ -5189,6 +5189,72 @@ bool CStaticFunctionDefinitions::GUIComboBoxSetItemText(CClientEntity& Entity, i
     return false;
 }
 
+int CStaticFunctionDefinitions::GUIComboBoxGetItemCount(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxGetItemCount(**iter))
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: RUN_CHILDREN is unnecessary in Get methods

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 5]
```diff
@@ -5189,6 +5189,72 @@ bool CStaticFunctionDefinitions::GUIComboBoxSetItemText(CClientEntity& Entity, i
...
+            return true;
+        }
+    }
+    
+    return false;
+}
+
+bool CStaticFunctionDefinitions::GUIComboBoxIsOpen(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxIsOpen(**iter))
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: As above, RUN_CHILDREN is unnecessary in Get methods

File: Client/gui/CGUIComboBox_Impl.cpp [hunk 6]
```diff
@@ -241,6 +241,11 @@ void CGUIComboBox_Impl::Clear(void)
     m_pWindow->setText(storedCaption);
 }
 
+bool CGUIComboBox_Impl::GetState(void)
```
Comment (qaisjp) [inline Client/gui/CGUIComboBox_Impl.cpp]: Should be renamed to IsOpen

**Review by qaisjp — CHANGES_REQUESTED** [review]

**Comment by patrikjuvonen** [comment]
@qaisjp @Jusonex What's the status on these reviews?

**Pull request merged**